cmd/config_options.go

```diff
@@ -15,51 +15,123 @@ limitations under the License.
 package cmd
 
 import (
+	"errors"
 	"fmt"
-	"log/slog"
+	"github.com/falcosecurity/falcoctl/pkg/options"
+	"github.com/falcosecurity/falcoctl/pkg/output"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"io"
 	"os"
+	"strings"
 
 	"github.com/creasty/defaults"
 	"github.com/falcosecurity/driverkit/validate"
 	"github.com/go-playground/validator/v10"
+	"github.com/pterm/pterm"
 )
 
 var validProcessors = []string{"docker", "kubernetes", "kubernetes-in-cluster", "local"}
 var aliasProcessors = []string{"docker", "k8s", "k8s-ic"}
-var configOptions *ConfigOptions
 
 // ConfigOptions represent the persistent configuration flags of driverkit.
 type ConfigOptions struct {
 	ConfigFile string
-	LogLevel   string `validate:"loglevel" name:"log level" default:"INFO"`
 	Timeout    int    `validate:"number,min=30" default:"120" name:"timeout"`
 	ProxyURL   string `validate:"omitempty,proxy" name:"proxy url"`
 	DryRun     bool
 
-	configErrors bool
+	// Printer used by all commands to output messages.
+	Printer *output.Printer
+	// Writer is used to write the output of the printer.
+	Writer   io.Writer
+	logLevel *options.LogLevel
+}
+
+func (co *ConfigOptions) initPrinter() {
+	logLevel := co.logLevel.ToPtermLogLevel()
+	co.Printer = output.NewPrinter(logLevel, pterm.LogFormatterColorful, co.Writer)
 }
 
 // NewConfigOptions creates an instance of ConfigOptions.
-func NewConfigOptions() *ConfigOptions {
-	o := &ConfigOptions{}
+func NewConfigOptions() (*ConfigOptions, error) {
+	o := &ConfigOptions{
+		Writer:   os.Stdout,
+		logLevel: options.NewLogLevel(),
+	}
+	o.initPrinter()
 	if err := defaults.Set(o); err != nil {
-		slog.With("err", err.Error(), "options", "ConfigOptions").Error("error setting driverkit options defaults")
-		os.Exit(1)
+		// Return ConfigOptions anyway because we need the logger
+		return o, err
 	}
-	return o
+	return o, nil
 }
 
 // Validate validates the ConfigOptions fields.
-func (co *ConfigOptions) Validate() []error {
+func (co *ConfigOptions) validate() []error {
 	if err := validate.V.Struct(co); err != nil {
-		errors := err.(validator.ValidationErrors)
-		errArr := []error{}
-		for _, e := range errors {
+		var errs validator.ValidationErrors
+		errors.As(err, &errs)
```

Comment:
Use `errors.As` as suggested by ide.
This kind of cleanup is done multiple times.